packages/crayons-core/src/components/nested-select/nested-select.tsx

```diff
@@ -0,0 +1,86 @@
+import { Component, Prop, Listen, h, EventEmitter, Event } from '@stencil/core';
+
+@Component({
+  tag: 'fw-nested-select',
+  styleUrl: 'nested-select.scss',
+  shadow: true,
+})
+export class NestedSelect {
+  private selections = [];
+  private selectedItems = {};
+  /**
+   * Options to display
+   */
+  @Prop() options = [];
+  /**
+   * Name of first level field
+   */
+  @Prop() name = '';
+  /**
+   * label
+   */
+  @Prop() label = '';
+  /**
+   * Initial value from first level choices
+   */
+  @Prop() value = '';
+  /**
+   * OptionValuePath referred from field
+   */
+  @Prop() optionValuePath = 'id';
+  /**
+   * OptionLabelPath referred from field
+   */
+  @Prop() optionLabelPath = 'value';
+  /**
+   * Function to return initialValues
+   */
+  @Prop() selectProps?: any;
+  /**
+   * Triggered when nested selection doesn't have choices
+   */
+  @Event() fwChange: EventEmitter;
+
+  @Listen('fwChange')
+  changed(event) {
+    if (!event.detail.meta) {
+      return;
+    }
+
+    this.selections[event.detail.level] = event.detail.meta.selectedOptions[0];
+    const itemsToRemove = this.selections.length - (event.detail.level + 1);
+    if (itemsToRemove > 0) {
+      this.selections = this.selections.slice(0, event.detail.level + 1);
+    }
+    this.getSelectedId(this.selections[event.detail.level], event.detail.name);
```

Comment:
Can we handle the scenario when "this.selections" can be empty here please

Reply:
Changed this code.